### PR TITLE
Update C changelog for C API 0.99.15

### PR DESCRIPTION
**Pull request by benjeffery**
I'm not sure what the stack allocation thing is? I searched PRs and Commit messages.

### CODE REVIEW

File: c/CHANGELOG.rst [hunk 1]
```diff
@@ -59,20 +62,26 @@
...
 
-- Add the TSK_LOAD_SKIP_REFERENCE_SEQUENCE option to load a table collection
+- Add the ``TSK_LOAD_SKIP_REFERENCE_SEQUENCE`` option to load a table collection
   without the reference sequence. Also add the TSK_CMP_IGNORE_REFERENCE_SEQUENCE
   option to compare two table collections without comparing their reference
   sequence. (:user:`clwgg`, :pr:`2019`, :issue:`1971`).
 
-- FIXME add features for virtual root, num_edges, stack allocation size etc
+- Add a "virtual root" to ``Tree`` arrays such as ``left_sib``, ``right_child`` etc.
+  The root inserted at the end of the array, has all real roots as it's children,
```
Comment (jeromekelleher) [inline c/CHANGELOG.rst:78]:
```suggestion
  The virtual root is appended to each array, has all real roots as its children,
```

File: c/CHANGELOG.rst [hunk 2]
```diff
@@ -59,20 +62,26 @@
...
+  but is not the parent of any node. Simplifies traversal algorithms.
+  (:user:`jeromekelleher`, :issue:`1691`, :pr:`1704`)
+
+- Add ``num_edges`` to ``tsk_tree_t`` to count the edges that define the topology of
+  the tree. (:user:`jeromekelleher`, :pr:`1704`)
```
Comment (jeromekelleher) [inline c/CHANGELOG.rst:83]:
```suggestion
  the tree. (:user:`jeromekelleher`, :pr:`1704`)
  
- Add the ``tsk_tree_get_size_bound`` function which returns an upper bound on the number of nodes reachable from 
  the roots of a tree. Useful for tree stack allocations (:user:`jeromekelleher`, :pr:`1704`).
```

**Review by jeromekelleher — APPROVED** [review]
LGTM, thanks for cleaning this up @benjeffery! I've added a few bits in the suggestions

File: c/CHANGELOG.rst (hunk 1 above)
Comment (benjeffery) [reply c/CHANGELOG.rst]: Done

File: c/CHANGELOG.rst (hunk 2 above)
Comment (benjeffery) [reply c/CHANGELOG.rst:83]: Done

**Pull request merged**